src/status_im2/contexts/chat/messages/content/text/view.cljs

```diff
@@ -1,11 +1,105 @@
 (ns status-im2.contexts.chat.messages.content.text.view
   (:require
+   [quo2.core :as quo]
+   [quo2.foundations.colors :as colors]
    [react-native.core :as rn]
-   [status-im.ui2.screens.chat.messages.message :as old-message]
-   [status-im2.contexts.chat.messages.link-preview.view :as link-preview]))
+   [status-im2.contexts.chat.messages.content.text.style :as style]
+   [status-im2.contexts.chat.messages.link-preview.view :as link-preview]
+   [utils.re-frame :as rf]))
+
+(defn mention-element
+  [from]
+  (rf/sub [:contacts/contact-name-by-identity from]))
+
+(defn render-inline
+  [acc {:keys [type literal destination]}]
+  (case type
+    ""
+    (conj acc literal)
+
+    "code"
+    (conj acc [rn/view {:style (merge style/block (style/code))} [quo/text {:weight :code} literal]])
+
+    "emph"
+    (conj acc [quo/text {:style {:font-style :italic}} literal])
+
+    "strong"
+    (conj acc [quo/text {:weight :bold} literal])
+
+    "strong-emph"
+    (conj acc
+          [quo/text
+           {:weight :bold
+            :style  {:font-style :italic}} literal])
+
+    "del"
+    (conj acc [quo/text {:style {:text-decoration-line :line-through}} literal])
+
+    "link"
+    (conj acc
+          [quo/text
+           {:style    {:color                (colors/theme-colors colors/primary-50 colors/primary-60)
+                       :text-decoration-line :underline}
+            :on-press #(rf/dispatch [:browser.ui/message-link-pressed destination])}
+           destination])
+
+    "mention"
+    (conj
+     acc
+     [rn/touchable-opacity
+      {:active-opacity 1
+       :on-press       #(rf/dispatch [:chat.ui/show-profile literal])
+       :style          (merge style/block {:background-color colors/primary-50-opa-10})}
+      [quo/text
+       {:weight :medium
+        :style  {:color (colors/theme-colors colors/primary-50 colors/primary-60)}}
+       [mention-element literal]]])
+
+    "status-tag"
+    (conj acc
+          [quo/text
+           {:style    {:color                (colors/theme-colors colors/primary-50 colors/primary-60)
+                       :text-decoration-line :underline}
+            :on-press #(rf/dispatch [:chat.ui/start-public-chat literal])}
+           "#"
+           literal])
+
+    (conj acc literal)))
+
+
+(defn render-block
+  [acc {:keys [type ^js literal children]}]
+  (case type
+
+    "paragraph"
+    (conj acc
+          (reduce
+           (fn [acc e] (render-inline acc e))
+           [quo/text]
+           children))
+
+    "blockquote"
+    (conj acc
+          [rn/view {:style style/quote}
+           [quo/text literal]])
+
+    "codeblock"
+    (conj acc
+          [rn/view {:style (merge style/block (style/code))}
+           [quo/text (.substring literal 0 (dec (.-length literal)))]])
+
+    acc))
+
+(defn render-parsed-text
+  [{:keys [content]}]
+  (reduce (fn [acc e]
```

Comment:
It's unnecessary to create an anonymous function. You can just use `(reduce render-inline)`.